pkg/parser/parser.go

```diff
@@ -87,6 +87,7 @@ func (p *Parser) Parse() *Program {
 func (p *Parser) parseProgram(scope *scope) *Program {
 	program := &Program{Statements: []Node{}}
 	p.advanceTo(0)
+	returnSeen := false
```

Comment:
rather than this `returnSeen` local var, you could add `hasReturn` to the `Program` struct, and also implement `HasReturn()` on it. Then you can set it and check it in a similar way to what I suggested on `parseBlockWithEndTokens()`.

Reply:
ok, great idea. done.